src/components/common/Header/cmp.tsx

```diff
@@ -4,14 +4,47 @@ import {
   Logo,
   NavbarLink,
   NavbarLinkList,
+  WalletPicker,
 } from '@aleph-front/aleph-core'
 import Link from 'next/link'
 import { StyledHeader, StyledButton, StyledNavbar } from './styles'
 import { ellipseAddress } from '@/helpers/utils'
 import { useHeader } from '@/hooks/pages/useHeader'
+import { useEffect, useRef, useState } from 'react'
 
 export const Header = () => {
-  const { theme, handleConnect, account, isOnPath } = useHeader()
+  const {
+    theme,
+    handleConnect,
+    account,
+    isOnPath,
+    displayWalletPicker,
+    setDisplayWalletPicker,
+    accountBalance,
+  } = useHeader()
+  const [isDivClicked, setDivClicked] = useState(false)
+  const divRef = useRef<HTMLDivElement | null>(null)
+
+  useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
+      if (divRef.current && !divRef.current.contains(event.target as Node)) {
+        setDisplayWalletPicker(false)
+      }
+    }
+
+    document.addEventListener('click', handleClickOutside)
+
+    return () => {
+      document.removeEventListener('click', handleClickOutside)
+    }
+  }, [])
+
+  const handleDisplayWalletPicker = (
+    event: React.MouseEvent<HTMLButtonElement>,
+  ) => {
+    event.stopPropagation()
+    setDisplayWalletPicker(!displayWalletPicker)
+  }
```

Comment:
Can you extract that into a reusable hook ? (or use [this one](https://usehooks-ts.com/react-hook/use-on-click-outside) ?)